doc_source/s3-encryption-client.rst

```diff
@@ -99,17 +110,27 @@ interface and requires two new parameters.
 Decryption
 ==========
 
-Downloading and decrypting an object requires only one additional parameter on
-top of ``GetObject``, and the client will detect the basic cipher options for you.
-Additional configuration options are passed through for decryption.
+Downloading and decrypting an object has four additional parameters, two of which are required, on top of the standard
+``GetObject`` parameters, two of which are required.  The client will detect the basic cipher options for you.
+
+* ``'@SecurityProfile'``:  If set to ‘V2’, only objects that are encrypted in V2-compatible
+   format can be decrypted. Setting this parameter  to ‘V2_AND_LEGACY’ also allows objects
+   encrypted in V1-compatible format to be decrypted. To support migration, set @SecurityProfile
+   to ‘V2_AND_LEGACY’.  Use ‘V2’ only for new application development.
+* ``'@MaterialsProvider'`` is a provider which handles generating a cipher key and initialization vector, as
+   well as encrypting your cipher key via AWS KMS.  The KMS key id is provided in the constructor of the provider.

```

Comment:
You shouldn't mention KMS-specific content here, as the materials provider is meant to be agnostic of the specific provider type, even if we currently only support KMS. We may add additional providers in the future, as it's set up to do so.

Nitpick, there's a double space after the period.

NOTE: Both these comments apply to the putObject section as well.